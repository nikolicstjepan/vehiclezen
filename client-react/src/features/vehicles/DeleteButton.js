import React, { useState } from "react";

export default function DeleteButton({ handleRemove }) {
  const [text, setText] = useState("Delete");

  const onClick = () => {
    if (text === "Delete") {
      return setText("Are you sure?");
    } else {
      handleRemove();
    }
  };

  return (
    <button className="btn btn-danger" onClick={onClick}>
      {text}
    </button>
  );
}
