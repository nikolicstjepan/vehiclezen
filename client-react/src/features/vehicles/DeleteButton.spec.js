import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import DeleteButton from "./DeleteButton";

describe("DeleteButton", () => {
  test("snapshot renders", () => {
    const component = renderer.create(<DeleteButton error="test" />);
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("has button with Delete text on it", () => {
    const component = shallow(<DeleteButton />);

    expect(component.find("button")).toHaveLength(1);
    expect(component.find("button").text()).toBe("Delete");
  });

  test("only changes button text on first button click", () => {
    const handleRemove = jest.fn(() => {});
    const component = shallow(<DeleteButton handleRemove={handleRemove} />);

    component.find("button").simulate("click");
    expect(component.find("button").text()).toBe("Are you sure?");
    expect(handleRemove).not.toHaveBeenCalled();
  });

  test("calls handleRemove function on second button click", () => {
    const handleRemove = jest.fn(() => {});
    const component = shallow(<DeleteButton handleRemove={handleRemove} />);

    component.find("button").simulate("click");
    component.find("button").simulate("click");
    expect(handleRemove).toHaveBeenCalled();
  });
});
