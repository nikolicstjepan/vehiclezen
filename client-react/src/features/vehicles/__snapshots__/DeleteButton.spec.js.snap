// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteButton snapshot renders 1`] = `
<button
  className="btn btn-danger"
  onClick={[Function]}
>
  Delete
</button>
`;
